app/Console/Commands/SetmoreAppointments.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\SetmoreAppointment;
+use Carbon\Carbon;
+use Config;
+use DB;
+use GuzzleHttp\Client;
+use Illuminate\Console\Command;
+
+class SetmoreAppointments extends Command {
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'SetmoreAppointments:Update';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Fetches training appointments from Setmore.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct() {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle() {
+        $setmoreAccessToken = null;
+        $setmoreAppointments = $setmoreServices = $setmoreStaff = [];
+        $from = Carbon::now()->subHours(2)->toDateString();
+        $to = Carbon::now()->addDays(90)->toDateString();
+        $client = new Client();
+        try {
+            $res = $client->request('GET', Config::get('setmore.endpoint') . '/o/oauth2/token?refreshToken=' . Config::get('setmore.api_key'));
+            $setmoreAuthResp = (string) $res->getBody();
+            $setmoreAuthRespA = json_decode($setmoreAuthResp, true);
+        } catch (ClientErrorResponseException $exception) {
+            echo "Unable to fetch access token from Setmore API: $exception";
+        }
+        if (isset($setmoreAuthRespA) && isset($setmoreAuthRespA['response'])) {
+            if ($setmoreAuthRespA['response'] && isset($setmoreAuthRespA['data']['token'])) {
+                $setmoreAccessToken = $setmoreAuthRespA['data']['token']['access_token'];
+            }
+        }
+        if ($setmoreAccessToken != null) {
+            $setmoreAppointments = $this->getSetmoreAppointments($setmoreAccessToken, $from, $to, true);
+            $setmoreServices = $this->getSetmoreServiceNames($setmoreAccessToken);
+            $setmoreStaff = $this->getSetmoreStaffNames($setmoreAccessToken);
+            DB::table('setmore')->truncate();
+            foreach ($setmoreAppointments as $setmoreAppointment) {
```

Comment:
Instead of truncating this table, can we fill this into a placeholder table, and then truncate and switch the data over into the table that the frontend interacts with? This would reduce downtime and prevent bad data if this fails in the middle of a pull

For more clarity this is the process I'm recommending:
* `[Setmore Data from API] -> [setmore_load table]`
  * Then everything succeeded?
    * `Truncate [setmore table]`
    * `[setmore_load table] -> [setmore table]`
  * Something went wrong?
    * Don't touch the setmore table
*  Irregardless of the result, `Truncate [setmore_load table]` at the end

I'm not sure if Laravel makes this easy to do, but it'll protect us from having random issues from popping up. This way we'll be able to tell if something is broken just based on whether the setmore data is updating or not. Additionally it'll reduce "downtime" (where there are no appointments or a partial listing of appointments listed in the DB) while updating

Reply:
Done - added `setmore_load` table and associated schema to truncate the table and cross-load the rows after the API pull has succeeded.